ttnn/cpp/ttnn/operations/data_movement/pad/pad.cpp

```diff
@@ -7,44 +7,144 @@
 #include "ttnn/common/constants.hpp"
 #include "ttnn/operations/core/core.hpp"
 #include "ttnn/run_operation.hpp"
-
+#include "ttnn/operations/data_movement/common/common.hpp"
 #include "ttnn/operations/data_movement/pad/device/pad_op.hpp"
 
 namespace ttnn::operations::data_movement {
 
 namespace {
 
-template <typename ShapeType>
+template <typename ArrayType>
+ttnn::SmallVector<uint32_t> array_to_vec(const ArrayType& arr) {
+    return ttnn::SmallVector<uint32_t>(arr.begin(), arr.end());
+}
+
+template <typename ArrayType>
+ttnn::SimpleShape array_to_simple_shape(const ArrayType& arr) {
+    return ttnn::SimpleShape(array_to_vec(arr));
+}
```

Comment:
can remove now that simple shape constructor is extended

Reply:
✅